tests/serialization/serialize_genome.cpp

```diff
@@ -9,55 +9,34 @@
 #define BOOST_TEST_MODULE Serialization test
 #include <boost/test/included/unit_test.hpp>
 
-std::string serialize(const NEAT::Genome &g)
+BOOST_AUTO_TEST_CASE(serialize_genome)
 {
-    std::ostringstream output_data;
+    NEAT::Parameters params;
+    NEAT::RNG rng;
 
-    {
-        cereal::JSONOutputArchive archive(output_data);
-        archive << g;
-    }
+    NEAT::Genome genome(42,
+                        2, 1, 3,
+                        NEAT::ActivationFunction::SIGNED_SIGMOID,
+                        NEAT::ActivationFunction::SIGNED_SIGMOID,
+                        params);
+    std::string serialized = genome.Serialize();
 
-    return output_data.str();
-}
+    NEAT::Genome genome_loaded;
+    genome_loaded.Deserialize(serialized);
 
-NEAT::Genome deserialize(const std::string &data)
-{
-    NEAT::Genome g;
-    std::istringstream input_data(data);
+    BOOST_TEST(genome == genome_loaded);
+
+    auto innov_db = NEAT::InnovationDatabase();
+
+    do
     {
-        cereal::JSONInputArchive archive(input_data);
-        archive >> g;
-    }
-    return g;
-}
+        genome.Mutate(false, SearchMode::BLENDED, innov_db, params, rng);
+    } while (innov_db.m_Innovations.size() < 100);
 
-BOOST_AUTO_TEST_CASE(serialize_genome)
-{
-    NEAT::Parameters params;
+    serialized = genome.Serialize();
+
+    NEAT::Genome genome_loaded2;
+    genome_loaded2.Deserialize(serialized);
 
-    NEAT::Genome g(42,
-            2, 1, 3,
-            NEAT::ActivationFunction::SIGNED_SIGMOID,
-            NEAT::ActivationFunction::SIGNED_SIGMOID,
-            params
-            );
-
-    const std::string serialized = serialize(g);
-    NEAT::Genome copy = deserialize(serialized);
-
-    BOOST_TEST(g.GetID() == copy.GetID());
```

Comment:
I would keep these tests as well, just so we test if we ever break the `operator==` for Genome. What do you think?

Reply:
Part of old tests restored